web/ui/react-app/src/api/Fetch.tsx

```diff
@@ -0,0 +1,75 @@
+import { Component } from 'react';
+import { fetchAPI } from './FetchAPI';
+
+export interface FetchState<T> {
+  data?: T;
+  error?: Error;
+}
+
+interface FetchProps {
+  url?: string;
+  urls?: string[];
```

Comment:
I think this should be more strongly typed. Could you just have a `urls` field? People who just want to fetch data at one url can just pass in `[url]`